marketing/utils.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Utility function for marketing app
+"""
+
+import smtplib
+import logging
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.template import Context
+
+from marketing.models import Subscriber, EmailTemplate, SentEmail
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger(__name__)
+
+
+def render_and_dispatch_email(template: EmailTemplate, subscriber: Subscriber):
+    """
+    Given an EmailTemplate and Subscriber, renders the email from
+    template and sends the email.
+    """
+    application = subscriber.user.betatestapplication
+    user = subscriber.user
+    context = Context({
+        "full_name": user.get_full_name(),
```

Comment:
When registering using the registration form, the user's name is stored in the profile model (`user.profile.full_name`) so this method doesn't return the full name from the correct place (make sure to prefetch it).